### PR TITLE
Added Interactive legend to ppc_plot bokeh

### LINKED ISSUE #1563: Show legend in ppc plot (backend bokeh)
## Tell us about it

xref https://github.com/arviz-devs/arviz/pull/1559#issuecomment-778762939

The matplotlib plots have legend, making it clear what each line/colour type represents. The bokeh plots don't seem to have them, and adding them would make the meaning of the plot clearer

### PR DESCRIPTION
**Pull request by Rishabh261998**
## Description
To fix #1563 
I have added an interactive legend to `ppc_plot` in bokeh.
Is the position of the legend is good enough?
Screenshot from 2021-03-06 01-10-25

Screenshot from 2021-03-06 01-10-35

## Checklist

- [x] Follows [official](https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md#pull-request-checklist) PR format
- [x] Includes a sample plot to visually illustrate the changes (only for plot-related functions)
- [ ] New features are properly documented (with an example if appropriate)?
- [x] Includes new or updated tests to cover the new feature
- [x] Code style  correct (follows pylint and black guidelines)
- [x] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/main/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

**Comment by Rishabh261998** [comment]
@OriolAbril could you look into this? Thanks

File: arviz/plots/backends/bokeh/ppcplot.py [hunk 1]
```diff
@@ -260,18 +284,32 @@ def plot_ppc(
...
 
+            legend_it.append(("{} predictive".format(group.capitalize()), all_scatter))
             ax_i.yaxis.major_tick_line_color = None
             ax_i.yaxis.minor_tick_line_color = None
             ax_i.yaxis.major_label_text_font_size = "0pt"
 
+        if legend:
+            legend = Legend(
+                items=legend_it,
+                location="center_right",
```
Comment (OriolAbril) [inline arviz/plots/backends/bokeh/ppcplot.py:308]: given that the legend is fixed, I would place it on the top left, which will generally be en empty corner for both kde and cdf plots. https://arviz-devs.github.io/arviz/examples/plot_ppc_cumulative.html https://arviz-devs.github.io/arviz/examples/plot_ppc.html

**Review by OriolAbril — APPROVED** [review]
Thanks!

File: arviz/plots/backends/bokeh/ppcplot.py (hunk 1 above)
Comment (Rishabh261998) [reply arviz/plots/backends/bokeh/ppcplot.py]: Done!
Screenshot from 2021-03-06 15-18-12

**Comment by Rishabh261998** [comment]
@OriolAbril could you please review this? Thanks

**Comment by canyon289** [comment]
@Rishabh261998 Thank you for this contribution.

Can a small test be added or amended  to cover this line?
https://app.codecov.io/gh/arviz-devs/arviz/compare/1602/diff#D1L313
image

**Comment by Rishabh261998** [comment]
@canyon289 for line 313?

File: arviz/plots/backends/bokeh/ppcplot.py [hunk 2]
```diff
@@ -260,18 +284,34 @@ def plot_ppc(
...
 
+        if legend:
+            legend = Legend(
+                items=legend_it,
+                location="top_left",
+                orientation="vertical",
+            )
+            ax_i.add_layout(legend)
+            if textsize is not None:
+                ax_i.legend.label_text_font_size = str(textsize) + "pt"
```
Comment (OriolAbril) [inline arviz/plots/backends/bokeh/ppcplot.py:313]: I would use an f-string here instead of adding strings

**Review by OriolAbril — COMMENTED** [review]
Regarding tests, I would make a copy of https://github.com/arviz-devs/arviz/blob/main/arviz/tests/base_tests/test_plots_bokeh.py#L880 without the parametrizations, name it test_plot_ppc_textsize and call plot_ppc using textsize

File: arviz/plots/backends/bokeh/ppcplot.py (hunk 2 above)
Comment (Rishabh261998) [reply arviz/plots/backends/bokeh/ppcplot.py]: @OriolAbril  Done!

**Review by OriolAbril — APPROVED** [review]

**Pull request merged**